gcp_variant_transforms/transforms/partition_variants_test.py

```diff
@@ -0,0 +1,130 @@
+    # Partition 2
+    variants.append(vcfio.Variant(reference_name='chr3', start=0))
+    variants.append(vcfio.Variant(reference_name='chr03', start=0))
+    expected_partitions[2] = variants[4:6]
+    # Partition 5
+    variants.append(vcfio.Variant(reference_name='6', start=0))
+    variants.append(vcfio.Variant(reference_name='06', start=0))
+    variants.append(vcfio.Variant(reference_name='chr06', start=0))
+    expected_partitions[5] = variants[6:9]
+    # Partition 9
+    variants.append(vcfio.Variant(reference_name='chr10', start=0))
+    variants.append(vcfio.Variant(reference_name='  chr10   ', start=0))
+    expected_partitions[9] = variants[9:11]
+    # Partition 21
+    variants.append(vcfio.Variant(reference_name='chr22', start=0))
+    variants.append(vcfio.Variant(reference_name='chr22    ', start=0))
+    expected_partitions[21] = variants[11:13]
+
+    # The rest of partitions will be empty
+    expected_partitions[3] = []
+    expected_partitions[4] = []
+    expected_partitions[6] = []
+    expected_partitions[7] = []
+    expected_partitions[8] = []
+    expected_partitions[10] = []
+    expected_partitions[11] = []
+    expected_partitions[12] = []
+    expected_partitions[13] = []
+    expected_partitions[14] = []
+    expected_partitions[15] = []
+    expected_partitions[16] = []
+    expected_partitions[17] = []
+    expected_partitions[18] = []
+    expected_partitions[19] = []
+    expected_partitions[20] = []
+    return variants, expected_partitions
+
+  def _get_nonstandard_variants(self):
+    # All these variants will not match to standard reference_name Reg Exp, thus
+    # they will all end up in partitions >= 22.
+    variants = []
+    expected_partitions = {}
+    # Partition 22
+    variants.append(vcfio.Variant(reference_name='NOT6', start=0))
+    variants.append(vcfio.Variant(reference_name='NOTch3', start=0))
+    expected_partitions[22] = variants[0:2]
+    # Partition 24
+    variants.append(vcfio.Variant(reference_name='NOTchr1', start=0))
+    variants.append(vcfio.Variant(reference_name='chr99', start=0))
+    expected_partitions[24] = variants[2:4]
+    # Partition 25
+    variants.append(vcfio.Variant(reference_name='NOT07', start=0))
+    variants.append(vcfio.Variant(reference_name='chr008', start=0))
+    variants.append(vcfio.Variant(reference_name='chr23', start=0))
+    variants.append(vcfio.Variant(reference_name='chrX', start=0))
+    variants.append(vcfio.Variant(reference_name='chrM', start=0))
+    expected_partitions[25] = variants[4:9]
+    # Partition 26
+    variants.append(vcfio.Variant(reference_name='NOTc4', start=0))
+    variants.append(vcfio.Variant(reference_name='chr0', start=0))
+    variants.append(vcfio.Variant(reference_name='chrY', start=0))
+    expected_partitions[26] = variants[9:12]
+    # The rest of partitions will be empty
+    expected_partitions[23] = []
+    return variants, expected_partitions
+
+  def test_partition_variants(self):
```

Comment:
This is still a bit difficult to read; consider the following approach using some python magic :D
```
expected_partitions = {}
expected_partitions[0] = [vcfio.Variant(reference_name='chr1', start=0),
                          vcfio.Variant(reference_name='CHR1', start=0)]
...etc
# do not add empty partitions as it can be error prone (e.g. what if you change the default #?)
return expected_partitions
```
In your test then:
```
variant_partitions = _get_standard_variant_partitions()
variant_partitions.update(_get_non_standard_variant_partitions())
variants = [variant for variant_list in variant_partitions for variant in variant_list]  # essentially flattens the lists
...
pipeline | Create(variants) 
...
for i in xrange(partitioner.get_num_partitions()):
  assert_that(partitions[i], equal_to(variant_partitions.get(i, [])), label=str(i))

```

Reply:
done.